src/Symfony/Bridge/Twig/Extension/AssetExtension.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Twig\Extension;
+
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Asset\Packages;
+
+/**
+ * Twig extension for the Symfony Asset component.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+class AssetExtension extends \Twig_Extension
+{
+    private $packages;
+
+    public function __construct(Packages $packages)
+    {
+        $this->packages = $packages;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFunctions()
+    {
+        return array(
+            new \Twig_SimpleFunction('asset_path', array($this, 'getAssetPath')),
```

Comment:
Even with what you propose I don't see a clear update path. OS bundles will keep using the `asset` function because they won't know if in your project you have enabled the new component and can start using `asset_path`. So when Symfony 3 comes they'll have to update to the new function name to stay compatible.

Besithes, I'm pretty sure that in 99.9% of the `asset` function calls out there in external bundles there are no packages and no use of the `absolute` or `version` params (or like the WebProfilerBundle, they removed all asset calls and changed them for inline base64-encoded images). So keeping the same name is the best option for them, they probably won't even have to bother about the change, the simple `asset('/path/to/my/bundle/foo.jpg')` call has the exact same syntax in the new layer if we keep the name.

Made a quick search here https://github.com/search?p=6&q=asset+NOT+url+extension%3Atwig&ref=searchresults&type=Code&utf8=%E2%9C%93 (unfortunately you can't search the parentheses symbol, which would narrow down the search a lot)